word_count.rb

```diff
@@ -0,0 +1,21 @@
+#Generates the count of each word in the file sorted by the maximum occured word first
+def wordCount
+  file = File.open("speech.txt")
+
+  words = file.read().downcase().split(/\W+/)
+  words_map = {}
+  words.map do |word|
+    if words_map.has_key?(word)
+      words_map[word] += 1
+    else
+      words_map[word] = 1
```

Comment:
`Hash.new` lets you set default values.
https://ruby-doc.org/core-1.9.3/Hash.html